server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -104,6 +92,29 @@ public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTyp
                 .getZonedDateTime());
     }
 
+    private Insert createSpecificUpdate(VacationUpdate vacationUpdate, Insert baseInsert) {
+        Insert insertWithSubject = addValue(baseInsert, CassandraVacationTable.SUBJECT, vacationUpdate.getSubject());
```

Comment:
What about generating a stream of `UnaryOperator<Insert>` then chain them with `reduce()` and `.andThen()`.
You would end up with a `UnaryOperator<Insert>` that you would apply on the first `Insert`.
The cool thing about that is that you can generate an Identity for fields that don't change, it would combine as well.